shell/src/main/java/org/crsh/lang/impl/groovy/closure/PipeLineClosure.java

```diff
@@ -192,7 +192,7 @@ else if (ret instanceof Iterable) {
         else {
           boolean use = true;
           for (Object value : closureOptions.values()) {
-            if (value == ret) {
+            if (value.equals(ret)) {
```

Comment:
This is equivalent to Object.equals().  It still checks for identity, so this buys nothing except for a more cluttered expression.